js-w2-hw.js

```diff
@@ -0,0 +1,129 @@
+// Flight booking fullname function
+
+function getFullname(firstName, lastName, useFormalName, gender) {
+  // Check if useFormalName is undefined
+  if (useFormalName === undefined) {
+    return "Error: Missing argument";
+  }
+
+  // Handle empty strings for firstName and lastName
+  if (!firstName && !lastName) {
+    return "Anonymous";
+  } else if (!firstName) {
+    firstName = "";
+  } else if (!lastName) {
+    lastName = "";
+  }
+
+  const fullName = `${firstName} ${lastName}`;
+
+  // Add formal title if useFormalName is true
+  if (useFormalName) {
+    if (gender === "male") {
+      return `Lord ${fullName}`;
+    } else if (gender === "female") {
+      return `Lady ${fullName}`;
+    }
+  }
+  return fullName;
+}
+
+// Call the function and assign the results to variables
+const fullname1 = getFullname("John", "Malkovich", true, "male");
+const fullname2 = getFullname("Sarah", "Conor", true, "female");
+const fullname3 = getFullname("Sarah", "Conor");
+const fullname4 = getFullname("", "", true, "male");
+
+// Log the variables to the console
+console.log(fullname1);
+console.log(fullname2);
+console.log(fullname3);
+console.log(fullname4);
+
+// EVENT APP
+
+// Array of weekday names
+function getEventWeekday(daysFromToday) {
+  const weekdays = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+
+  // Get today's day index (0-6, where 0 is Sunday and 6 is Saturday)
+  const today = new Date().getDay();
+
+  // Calculate the event day index
+  const eventDay = (today + daysFromToday) % 7;
+
+  // Return the weekday name
+  return weekdays[eventDay];
+}
```

Comment:
Good job here. I would only add checks for undefined values, or what happens if you provide a value that is not a number for example.